omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/wrapper/openmrs1_8/Module1_8.java

```diff
@@ -0,0 +1,38 @@
+package org.openmrs.module.webservices.rest.web.v1_0.wrapper.openmrs1_8;
+
+import org.openmrs.BaseOpenmrsMetadata;
+import org.openmrs.module.Module;
+
+import java.io.Serializable;
+
+/**
+ * Created by user on 22.05.16.
```

Comment:
I presume you introduced this so you could use the MetadataDelegatingXyz class. You should comment in the class-level javadoc why you're doing this.

That said I think it's quite a hack, and I don't think that shortcut is okay.

Can you instead have the resource extends BaseDelegatingReadableResource or BaseDelegatingResource? (I don't think these have any bounds on the class you'd delegate to.

(Of course, I'm not even sure you need one the abstract base classes here. Couldn't you just directly implement Retrievable and Listable? And if you're going this path you could probably also implement Converter<Module>.)